parsons/actblue/actblue.py

```
@@ -0,0 +1,132 @@
+import logging
+import time
+import urllib.request
+from parsons.utilities.api_connector import APIConnector
+from parsons.utilities import check_env
+from parsons.etl import Table
+
+
+logger = logging.getLogger(__name__)
+
+class ActBlue(object):
+    """
+    Instantiate class.
+
+       `Args:`
+            actblue_client_uuid: str
+                The ActBlue provided Client UUID. Not required if ``ACTBLUE_CLIENT_UUID`` env variable set.
+            actblue_client_secret: str
+                The ActBlue provided Client Secret. Not required if ``ACTBLUE_CLIENT_SECRET`` env variable set.
+
+        For instructions on how to generate a Client UUID and Client Secret set, visit https://secure.actblue.com/docs/csv_api#authentication.
+    """
+
+    def __init__(self, actblue_client_uuid=None, actblue_client_secret=None):
+        self.actblue_client_uuid = check_env.check('ACTBLUE_CLIENT_UUID', actblue_client_uuid)
+        self.actblue_client_secret = check_env.check('ACTBLUE_CLIENT_SECRET', actblue_client_secret)
```

could we / should we raise an error if these are missing?

---

yes! Just added a TODO to the code; we will add a 'raise error' line before the final MR.